crypto11.go

```diff
@@ -285,13 +285,16 @@ func Configure(config *PKCS11Config) (*pkcs11.Ctx, error) {
 // Note that if CRYPTO11_CONFIG_PATH is set in the environment,
 // configuration will be read from that file, overriding any later
 // runtime configuration.
-func ConfigureFromFile(configLocation string) (*pkcs11.Ctx, error) {
+func ConfigureFromFile(configLocation string) (ctx *pkcs11.Ctx, err error) {
 	file, err := os.Open(configLocation)
 	if err != nil {
 		log.Printf("Could not open config file: %s", configLocation)
 		return nil, err
 	}
-	defer file.Close()
+	defer func() {
+		err = file.Close()
```

Comment:
That doesn't look right. Suppose the file contents are malformed. `Decode` below will return a non-nil error and try to return it, but `file.Close()` will return `nil`. The result will be that `ConfigureFromFile` returns `nil` despite an error having occurred.

See https://play.golang.org/p/rzTxJM85e0c for a reduced example.

Reply:
Yikes, good spot. I'll take a look for other places I may have introduced the same bug with this pattern. Thanks :-)